db/version_set.cc

```diff
@@ -1125,6 +1128,22 @@ VersionSet::Finalize(Version* v)
         // consider this level
         if (compact_ok)
         {
+            size_t grooming_trigger;
+
+            // which grooming trigger point?  based upon how long
+            //  since last compaction on this level
+            //   - less than 10 minutes?
+            if (micros_now<(m_CompactionStatus[level].m_LastCompaction + (1000000 * 60 *10)))
+                grooming_trigger=config::kL0_GroomingTrigger;
+
+            //   - less than 20 minutes?
+            else if (micros_now<(m_CompactionStatus[level].m_LastCompaction + (1000000 * 60 *20)))
```

Comment:
I recommend defining (1000000 * 60 *10) and (1000000 * 60 *20) as constants (probably in db/dbformat.h, along with kL0_GroomingTrigger10min and kL0_GroomingTrigger20min)